src/lib.rs

```diff
@@ -720,7 +735,12 @@ extern "C" fn handle_message(
                 jsep: unsafe { JanssonValue::from_raw(jsep) },
             };
             janus_info!("Queueing signalling message on {:p}.", sess.handle);
-            MESSAGE_CHANNEL.get().unwrap().send(msg).ok();
+            let mut message_count = MESSAGE_COUNTER.lock().unwrap();
+            let senders = MESSAGE_SENDERS.get().unwrap();
+            let sender = &senders[*message_count % senders.len()];
+            sender.send(msg).ok();
+            *message_count += 1;
```

Comment:
This round-robin approach is easy but might have some edge cases depending on the variance in how long messages take to process. If you got a slow message followed by many fast messages, you might push the slow message to some queue, push a fast message to all the other queues, and when you got back to the first queue, it would still be blocked, but you would push a fast message to it anyway (which would then have to wait) instead of using one of the others.

Reply:
yeah good point, hard to know how important this is to worry about without doing detailed analysis of wait times in production.

Reply:
i suspect that if someone joins a room with a few people, the N^2 nature of it just means all these threads will be slammed in parallel with the SDP hotspot until they flush through, so smarter queueing won't make a difference but hard to know for sure without measuring. if that happened we'd probably want to dispatch to threads based upon message type.